File: Regular Expressions/Code/chapter6/6-13/recipe6-13.pl
#!/usr/bin/perl -w
use strict;

open ( FILE, $ARGV[0] ) || die "Cannot open file!";

while (<FILE>)
{ 
	my $line = $_;
	$line =~ s/(CREATE\s+PROCEDURE\s+)(?!\[?\w+\]?\.)(\w+)/$1\[dbo\].\[$2\]/g;
	print $line;
}

close( FILE );

File: Regular Expressions/Code/chapter6/6-13/recipe6-13.py
#!/usr/bin/python
import re
import sys

nargs = len(sys.argv)

if nargs > 1:

	myfile = sys.argv[1]
	output = open( myfile )
	regex = re.compile( r'(CREATE\s+PROCEDURE\s+)(?!\[?[A-Z]+\]?\.)(\w+)', re.I )
	lines = output.readlines()
	for line in lines:
		formatted = regex.sub( r'\1[dbo].[\2]', line )
		print formatted,

	output.close()

else:
	print 'Supply a parameter!',
